lexos/static/js/scripts_scrub.js

```diff
@@ -1,267 +1,309 @@
-          var selection = $('#allTags option:selected').val()
-          $('#tagTable').empty().remove()
-          var t = '<table id="tagTable" class="table table-condensed table-striped table-bordered"></table>'
-          $('#xmlModalBody').append(t)
-          var select = '<select id="allTags" style="margin-top:3px;margin-right:5px;">'
-          select += '<option value="remove-tag,allTags">Remove Tag Only</option>'
-          select += '<option value="remove-element,allTags">Remove Element and All Its Contents</option>'
-          select += '<option value="replace-element,allTags">Replace Element and Its Contents with Attribute Value</option>'
-          select += '<option value="leave-alone,allTags">Leave Tag Alone</option>'
-          select += '</select>'
-          select += '<button id="set-tags-button" type="button" class="btn btn-primary"">Set All</button>'
-          $('#tagTable').append('<thead><tr><th>Element</th><th>Action</th><th>' + select + '</th></tr></thead>')
-          $('#tagTable').append('<tbody></tbody>')
-		      $('#tagTable tbody').append(response)
-          $('#xmlModalStatus').remove()
-          $("#allTags option[value='"+selection+"']").prop("selected", true)
-	    		var value = $('#myselect option:selected').val()
-	    		var text = $('#myselect option:selected').text()
-        },
-        error: function (jqXHR, textStatus, errorThrown) {
-          console.log('Error: ' + errorThrown)
-        }
-      })
+  $('#formAction').val(action)
+  const formData = new FormData($('form')[0])
+
+  $.ajax({
+    url: '/doScrubbing',
+    type: 'POST',
+    processData: false, // important
+    contentType: false, // important
+    data: formData,
+    error: function (jqXHR, textStatus, errorThrown) {
+      $('#error-modal-message').html('Lexos could not apply the scrubbing actions.')
+      $('#error-modal').modal()
+      console.log('bad: ' + textStatus + ': ' + errorThrown)
     }
+  }).done(function (response) {
+    response = JSON.parse(response)
+    $('#preview-body').empty()
+    $.each(response['data'], function () {
+      const fileName = $(this)[1]
+      const fileContents = $(this)[3]
+      const fieldSet = $('<fieldSet></fieldSet>')
+      // CSS truncates the document name
+      fieldSet.append('<legend class="has-tooltip" style="color:#999; width:90%;margin: auto; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;">' + fileName + '</legend>')
+      fieldSet.append('<div class="filecontents">' + fileContents + '</div>') // Keep this with no whitespace!
+      $('#preview-body').append(fieldSet)
+      $('#status-prepare').css({'visibility': 'hidden'})
+    })
   })
+}
+
+/**
+ * Function to handle the chevron drop down button rotate animation by toggling
+ * the class so the appropriate CSS applies.
+ * @returns {void} - returns nothing
+ */
+function rotateChevron () {
+  $(this).find('span').toggleClass('down')
+  $(this).next().collapse('toggle')
```

Comment:
What does `this` means? I think you can just pass that in as a parameter?

Reply:
`this` is the legend which contains a chevron drop down button that the user clicks. I'm not sure that it can just be passed as this click function is tied to these legends by class name, and passing that as an argument instead of using `this` will cause all of the drop downs to open and chevrons to rotate when only one is clicked.

Reply:
I think you can just pass the `currentTarget` into this function.

Read this: https://api.jquery.com/event.currentTarget/

Also, please remove every single `this` that is in this code.

Reply:
The only example on that page uses an anonymous function and passes `event` as a parameter and is then able to do `event.currentTarget`. How do I do the same with a normal function?

Reply:
What is the justification for removing `$(this)`? Are there places where it generates scoping issues?

Reply:
The justification for removing `this` is to create robust functions. A function should does what is promise to do wherever it is invoked. If you use `this` in a function it is forced to be invoked in a specific scope. It creates robustness issue (we cannot reuse the function in other context), readability issue (people have to understand where the function is invoked to understand what the function does) and possible bugs (people might invoked the function in other context and hope it would 'handle the chevron drop down button rotate animation'). 

This is considered a good practice to avoid `this` outside of the context of a class see this eslint rule: https://eslint.org/docs/rules/no-invalid-this

Reply:
Hmmm... I need to evaluate this when I have more time. jQuery `$(this)` is not the same as Javascript `this`, so I am not sure that it shares the same scoping issues.

Reply:
It probably does, since it depends on the environment of the outer scope.

Reply:
```js
function rotateChevronOfLegend (legend) { 
  $(lengend).find('span').toggleClass('down') 
  $(lengend).next().collapse('toggle') 
}
```

when this is invoked:
```
$(stuff).on("click", function(e){rotateChevronOfLegend(e.currentTarget)})
```

Reply:
Ah, I see, this makes much more sense now. This actually fixed a problem I was having with the rotate animation.